### ISSUE TITLE
Emit diagnostics for unsupported comparisons involving non-fixed-length tuples

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#1741]
Pyright, mypy and pyrefly all emit diagnostics on all four comparisons here, but ty currently does not emit any diagnostics on this snippet:

```py
def f(
    a: tuple[int, ...],
    b: tuple[str, ...],
    c: tuple[str]
):
    a < b
    b < a
    a < c
    c < a
```

**Comment by AlexWaygood** [comment]
we already emit diagnostics if both operands are _fixed_-length tuples, e.g.

```py
def f(
    x: tuple[int, int],
    y: tuple[str]
):
    x < y  # error: [unsupported-operator]
```
